components/canvas/webgl_thread.rs

```diff
@@ -2163,107 +2452,94 @@ impl WebGLImpl {
             };
         }
-        let mut num_active_attribs = [0];
-        unsafe {
-            gl.get_program_iv(
-                program.get(),
-                gl::ACTIVE_ATTRIBUTES,
-                &mut num_active_attribs,
-            );
-        }
-        let active_attribs = (0..num_active_attribs[0] as u32)
+        let num_active_attribs =
+            unsafe { gl.get_program_parameter_i32(program.glow(), gl::ACTIVE_ATTRIBUTES) };
+        let active_attribs = (0..num_active_attribs as u32)
             .map(|i| {
                 // FIXME(nox): This allocates strings sometimes for nothing
                 // and the gleam method keeps getting ACTIVE_ATTRIBUTE_MAX_LENGTH.
-                let (size, type_, name) = gl.get_active_attrib(program.get(), i);
+                let active_attribute =
+                    unsafe { gl.get_active_attribute(program.glow(), i) }.unwrap();
+                let name = &active_attribute.name;
                 let location = if name.starts_with("gl_") {
-                    -1
+                    None
                 } else {
-                    gl.get_attrib_location(program.get(), &name)
+                    unsafe { gl.get_attrib_location(program.glow(), name) }
                 };
                 ActiveAttribInfo {
-                    name: from_name_in_compiled_shader(&name),
-                    size,
-                    type_,
+                    name: from_name_in_compiled_shader(name),
+                    size: active_attribute.size,
+                    type_: active_attribute.atype,
                     location,
                 }
             })
             .collect::<Vec<_>>()
             .into();
 
-        let mut num_active_uniforms = [0];
-        unsafe {
-            gl.get_program_iv(program.get(), gl::ACTIVE_UNIFORMS, &mut num_active_uniforms);
-        }
-        let active_uniforms = (0..num_active_uniforms[0] as u32)
+        let num_active_uniforms =
+            unsafe { gl.get_program_parameter_i32(program.glow(), gl::ACTIVE_UNIFORMS) };
+        let active_uniforms = (0..num_active_uniforms as u32)
             .map(|i| {
                 // FIXME(nox): This allocates strings sometimes for nothing
                 // and the gleam method keeps getting ACTIVE_UNIFORM_MAX_LENGTH.
-                let (size, type_, mut name) = gl.get_active_uniform(program.get(), i);
-                let is_array = name.ends_with("[0]");
+                let mut active_uniform =
+                    unsafe { gl.get_active_uniform(program.glow(), i) }.unwrap();
+                let is_array = active_uniform.name.ends_with("[0]");
                 if is_array {
                     // FIXME(nox): NLL
-                    let len = name.len();
-                    name.truncate(len - 3);
+                    let len = active_uniform.name.len();
+                    active_uniform.name.truncate(len - 3);
                 }
```

Comment:
This could maybe just be `trim_end_matches()` or some sort of `unwrap_or_else` combined with `strip_suffix`?